### DISCUSSION TITLE
Developing extensions for writing GAML in other text editors

### DISCUSSION THREAD

**Discussion by minhduc0711** [gama-platform/gama#42] [Ideas]
Hello guys, the other day @RoiArthurB mentioned the possibility of creating some extensions that allow writing GAML with cool features like syntax highlighting, autocompletion, and linting in other text editors (e.g. VSCode, Sublime, Vim...).
Such an extension would be really useful to me, so I'd love to help out!

This might be a nice starting point https://github.com/svenefftinge/xtext-lsp-workshop

**Comment by RoiArthurB** [comment]
Ping @hqnghi88

**Reply by hqnghi88** [reply]
bash: ping: command not found

**Comment by RoiArthurB** [comment]
Here's a more up-to-date example : https://github.com/cdietrich/xtext-languageserver-example

**Reply by RoiArthurB** [reply]
@minhduc0711 I just saw that, since XText 2.11 it should be natively compatible with lsp https://github.com/eclipse/xtext-core/wiki/Language-Server-Protocol 🤔🤔🤔
GAMA use the 2.24 version, so it should be easy to do 🤷🤷🤷

**Reply by minhduc0711** [reply]
Fyi, here is the [accompanying article](https://blogs.itemis.com/en/integrating-xtext-language-support-in-visual-studio-code) for the example repo you linked above.

**Comment by hqnghi88** [comment]
Hello, you two guys, can you update your work, as Arthur said you have try something and too many error with gaml xtext

**Reply by minhduc0711** [reply]
We tried some stuff in parallel, so I'm not sure what went wrong on Arthur's machine.

On my side, I followed this tutorial https://www.eclipse.org/Xtext/documentation/340_lsp_support.html, but I suppose I am having some dependency issues when running the LSP:
```
Caused by: java.lang.ClassNotFoundException: org.eclipse.lsp4j.services.LanguageServer
```
So perhaps I will need a few days to figure something out.

**Reply by minhduc0711** [reply]
One more thing, please correct these two facts if I am wrong: 
* The GAML syntax is defined entirely in `msi.gama.lang.gaml`
* The fancy editing features (e.g. code completion) is defined entirely in `ummisco.gama.ui.modeling`

**Reply by hqnghi88** [reply]
> We tried some stuff in parallel, so I'm not sure what went wrong on Arthur's machine.
> 
> On my side, I followed this tutorial https://www.eclipse.org/Xtext/documentation/340_lsp_support.html, but I suppose I am having some dependency issues when running the LSP:
> 
> ```
> Caused by: java.lang.ClassNotFoundException: org.eclipse.lsp4j.services.LanguageServer
> ```
> 
> So perhaps I will need a few days to figure something out.

OK


> One more thing, please correct these two facts if I am wrong:
> 
> * The GAML syntax is defined entirely in `msi.gama.lang.gaml`
> * The fancy editing features (e.g. code completion) is defined entirely in `ummisco.gama.ui.modeling`

Yes you are totally right
